src/toggle/toggle.tsx

```diff
@@ -0,0 +1,161 @@
+import { useCallback, useEffect, useRef, useState } from "react";
+import {
+    Container,
+    Input,
+    Label,
+    // SubLabel,
+    TextContainer,
+} from "./toggle.styles";
+import { ToggleProps } from "./types";
+import { ToggleIcon, ToggleIconType } from "../shared/toggle-icon/toggle-icon";
+
+export const Toggle = ({
+    type = "checkbox",
+    indicator,
+    checked,
+    styleType = "default",
+    children,
+    // subLabel,
+    disabled,
+    error,
+    name,
+    id,
+    className,
+    "data-testid": testId,
+    onChange,
+}: ToggleProps) => {
+    // =============================================================================
+    // CONST, STATE, REF
+    // =============================================================================
+    const [selected, setSelected] = useState<boolean>(checked);
+    const inputRef = useRef<HTMLInputElement>();
+    const generatedInputId = id ? `${id}-input` : crypto.randomUUID();
```

Comment:
can make use of lazy initial state to generate a fixed id for the lifecycle of the component

`const [state, setState] = useState(() => crypto.randomUUID())`

Reply:
in that case won't this work too? 
```
const [inputId] = useState<string>(id ? `${id}-input` : crypto.randomUUID());
```

Reply:
hmm yes. although need to be careful to keep `id` up to date.

the main difference is in the first one, the function only gets called once, whereas the second one will generate uuid on every rerender (just that the new value is not saved)

how about
```
const [uuid] = useState(() => crypto.randomUUID())
const generatedInputId = id ? `${id}-input` : uuid;
```

Reply:
Ok this is better. will do that